### PR TITLE
Add usage examples to throwsA and make other documentation refer to it

### PR DESCRIPTION
**Pull request by jamesderlin**
* Add examples to the `throwsA` documentation.  This is a frequent
  source of confusion on StackOverflow.
* Make the various top-level `throws...` matchers refer to the
  `throwsA` documentation for details.

Fixes https://github.com/dart-lang/test/issues/1447.

### CODE REVIEW

**Comment by jamesderlin** [comment]
Ping? (Maybe @srawlins ?)

**Comment by jamesderlin** [comment]
Anybody? (@natebosch ?)

File: pkgs/test_api/lib/src/frontend/throws_matcher.dart [hunk 1]
```diff
@@ -33,6 +33,30 @@ const Matcher throws = Throws();
 /// In all three cases, when an exception is thrown, this will test that the
 /// exception object matches [matcher]. If [matcher] is not an instance of
 /// [Matcher], it will implicitly be treated as `equals(matcher)`.
+///
+/// Examples:
+/// ```dart
+/// void functionThatThrows() => throw UnsupportedError('Unsupported');
```
Comment (natebosch) [inline pkgs/test_api/lib/src/frontend/throws_matcher.dart:39]: [nit] Testing behavior of `*Exception` being thrown should hopefully be more common than testing that a `*Error` is thrown. Can we maybe use some subtype of exception, or just a fake `SomeException` type?

Comment (jamesderlin) [reply pkgs/test_api/lib/src/frontend/throws_matcher.dart]: Fair point.  Done.

**Review by natebosch — APPROVED** [review]

**Pull request merged**